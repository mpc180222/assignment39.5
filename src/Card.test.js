import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Card from "./Card";

it("should render without crashing", function () {

  render(<Card />);

})

it('should match snapshot', () => {
    const imgUrl = 'https://images.unsplash.com/photo-1660663577778-c45295fc2dec?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80'

    const { asFragment } = render(<Card caption = "a nice picture" src= {imgUrl} currNum = {1} totalNum = {1} />)
    expect(asFragment()).toMatchSnapshot();


})