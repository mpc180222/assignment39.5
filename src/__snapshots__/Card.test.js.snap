// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      a nice picture
    </h4>
    <img
      alt="a nice picture"
      class="Card-image"
      src="https://images.unsplash.com/photo-1660663577778-c45295fc2dec?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80"
    />
    <small
      class="Card-small"
    >
      Image 1 of 1.
    </small>
  </div>
</DocumentFragment>
`;
